molo/profiles/models.py

```diff
@@ -80,10 +123,39 @@ class UserProfile(models.Model):
         null=True)
 
     mobile_number = PhoneNumberField(blank=True, null=True, unique=False)
+    security_question_answers = models.ManyToManyField(
+        SecurityQuestion,
+        through="SecurityAnswer"
+    )
 
 
 @receiver(post_save, sender=User)
 def user_profile_handler(sender, instance, created, **kwargs):
     if created:
         profile = UserProfile(user=instance)
         profile.save()
+
+
+class SecurityAnswer(models.Model):
+    user = models.ForeignKey(UserProfile)
+    question = models.ForeignKey(SecurityQuestion)
```

Comment:
This relationship is a bit of a problem for me - it essentially means that admins actions have a direct bearing on whether or not a user is able to recover their password. In other words, if an admin decides to change, update or delete a security question, the the answer that the user has saved is potentially now no longer relevant.

The one one way of going around this problem is to store the question text on the answer instead of the link to the Security Question. That way when asking the user for their answer, you're asking them for the question "as it was on the day of registration" and not what the current question is.